biomaj/bank.py

```diff
@@ -525,10 +527,36 @@ def save_session(self):
 
         self.bank = self.banks.find_one({'name': self.name})
 
+    def format_release(self, release, reverse=False):
+        """
+        Check the release does not contains any dot '.' character that should make MongoDB complain
+
+        :param release: release number to check
+        :type release: str
+        :param reverse: Do the reverse replacement
+        :type reverse: bool
+        :return: Release string with dot '.' replaced with character set in config (release.replace)
+        :rtype: str
+        """
+        if not self.config.get('release.replacedot'):
```

Comment:
maybe indeed should fix pending storage....  I have to think about this

Le mer. 27 avr. 2016 à 11:04, Emmanuel Quevillon <notifications@github.com>
a écrit :

> In biomaj/bank.py
> <https://github.com/genouest/biomaj/pull/44#discussion_r61225370>:
>
> > @@ -525,10 +527,36 @@ def save_session(self):
> >
> >          self.bank = self.banks.find_one({'name': self.name})
> >
> > +    def format_release(self, release, reverse=False):
> > +        """
> > +        Check the release does not contains any dot '.' character that should make MongoDB complain
> > +
> > +        :param release: release number to check
> > +        :type release: str
> > +        :param reverse: Do the reverse replacement
> > +        :type reverse: bool
> > +        :return: Release string with dot '.' replaced with character set in config (release.replace)
> > +        :rtype: str
> > +        """
> > +        if not self.config.get('release.replacedot'):
>
> Checking other module code, setting the release in the database with the
> 'transformed' value (6.10 -> 6#10) will lead to change other code calling
> xxx.get('release') for example sessions.get_release_directory() which
> build the release dir to be created on disk.
> This means that the directory name will be then saved as
> /path/to/datadir/bank/6#10 :(
> Another solution would to change the way pending sessions are stored :
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/genouest/biomaj/pull/44/files/d235f544d677e718229ab04b92229485fa50ca28#r61225370>
>